lib/tool/datafileviewxml.php

```diff
@@ -187,48 +184,87 @@ protected static function getFiles($path)
+     * @throws \Exception
 	 */
 	public static function parseFile(File $file)
 	{
 		$xmlParser = new \CDataXML();
 		$xmlParser->Load($file->getPath());
 		$xmlArray = $xmlParser->getArray();
-		$record = new Record();
-		$record->setXmlId($xmlArray["data"]["#"]["xml_id"][0]["#"]);
-		if ($xmlArray["data"]["@"]["deleted"])
-		{
-			$record->setDeleteMark(true);
-		}
-		else
-		{
-			if ($xmlArray["data"]["#"]["field"])
-			{
-				$fields = static::parseFields($xmlArray["data"]["#"]["field"]);
-				$record->addFields($fields);
-			}
+        if(is_array($xmlArray["data"]["#"]))
+        {
+            $record = new Record();
+            $record->setXmlId($xmlArray["data"]["#"]["xml_id"][0]["#"]);
+            if ($xmlArray["data"]["@"]["deleted"])
+            {
+                $record->setDeleteMark(true);
+            }
+            else
+            {
+                if ($xmlArray["data"]["#"]["field"])
+                {
+                    $fields = static::parseFields($xmlArray["data"]["#"]["field"]);
+                    $record->addFields($fields);
+                }
 
-			if ($xmlArray["data"]["#"]["dependency"])
-			{
-				$links = static::parseLinks($xmlArray["data"]["#"]["dependency"]);
-				$record->setDependencies($links);
-			}
+                if ($xmlArray["data"]["#"]["dependency"])
+                {
+                    $links = static::parseLinks($xmlArray["data"]["#"]["dependency"]);
+                    $record->setDependencies($links);
+                }
 
-			if ($xmlArray["data"]["#"]["reference"])
-			{
-				$links = static::parseLinks($xmlArray["data"]["#"]["reference"]);
-				$record->setReferences($links);
-			}
+                if ($xmlArray["data"]["#"]["reference"])
+                {
+                    $links = static::parseLinks($xmlArray["data"]["#"]["reference"]);
+                    $record->setReferences($links);
+                }
 
-			if ($xmlArray["data"]["#"]["runtime"])
-			{
-				$runtimes = static::parseRuntimes($xmlArray["data"]["#"]["runtime"]);
-				$record->setRuntimes($runtimes);
-			}
-		}
+                if ($xmlArray["data"]["#"]["runtime"])
+                {
+                    $runtimes = static::parseRuntimes($xmlArray["data"]["#"]["runtime"]);
+                    $record->setRuntimes($runtimes);
+                }
+            }
 
-		return $record;
+            return $record;
+        }
+        else
+        {
+            throw new \Exception(Loc::getMessage('INTERVOLGA_MIGRATO.INCORRECT_XML_FILE',
+                [
+                    '#FILE_NAME#' => $file->getName(),
+                    '#FILE_CATALOG#' => $file->getDirectoryName(),
+                ]
+            ));
+        }
 	}
 
+    /**
+     * @param \Bitrix\Main\IO\File $fileSystemEntry
+     *
+     * @return bool
+     * @throws \Exception
+     */
+    protected static function isCorrectFile($fileSystemEntry)
+    {
+        $name = strtolower($fileSystemEntry->getName());
```

Comment:
Тут и ниже в этом методе для строковых функций лучше использовать их mb_ аналоги
```suggestion
        $name = mb_strtolower($fileSystemEntry->getName());
```